packages/block-library/src/latest-posts/edit.native.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * External dependencies
+ */
+import { TouchableWithoutFeedback, View, Text } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { compose, withPreferredColorScheme } from '@wordpress/compose';
+import { withDispatch } from '@wordpress/data';
+import { coreBlocks } from '@wordpress/block-library';
+import { __ } from '@wordpress/i18n';
+import { postList as icon } from '@wordpress/icons';
+import { InspectorControls } from '@wordpress/block-editor';
+import { fetchRequest } from 'react-native-gutenberg-bridge';
+import {
+	Icon,
+	PanelBody,
+	ToggleControl,
+	SelectControl,
+	RangeControl,
+	QueryControls,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+import { MIN_EXCERPT_LENGTH, MAX_EXCERPT_LENGTH } from './constants';
+
+class LatestPostsEdit extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			categoriesList: [],
+		};
+	}
+
+	componentDidMount() {
+		this.isStillMounted = true;
+		this.fetchRequest = fetchRequest( '/wp/v2/categories' )
+			.then( ( categoriesList ) => {
+				if ( this.isStillMounted ) {
+					if ( typeof categoriesList === 'string' ) {
```

Comment:
Android returns a string and iOS returns an object as a result of `fetchRequest`. I debated using a Platform check instead of a type check but thought the type check was safer and more descript.

Reply:
This sounds reasonable. Probably out of scope for this PR, but I'm curious why this platform divergence exists.

I see for Android, we convert the response to a string here:

https://github.com/wordpress-mobile/WordPress-Android/blob/ddb3be8823733e4d6305e63978e8f8353b7c4528/WordPress/src/main/java/org/wordpress/android/ui/posts/reactnative/ReactNativeRequestHandler.kt#L82

and for iOS, we seem to have an object response from here:

https://github.com/wordpress-mobile/WordPressKit-iOS/blob/19a2f186ef6e08e95c4678fd2af487727f3da9a3/WordPressKit/WordPressComRestApi.swift#L203

which is passed along without unwrapping here:

https://github.com/wordpress-mobile/WordPress-iOS/blob/837fbd067d0af8162fa0e21af499a4e469c50b25/WordPress/Classes/ViewRelated/Gutenberg/GutenbergNetworking.swift#L26

cc: @mchowning and @etoledom :wave: :smiley:  Do you know if there is a technical reason the types differ by platform for the `fetchRequest` resolved promise?

Reply:
I can't remember the specific reason Android is using a `String` but there wasn't a technical reason we couldn't pass an object, it was related to it just being much easier to pass the response across the bridge as a String. 

Looking back at it now, I wonder if there is a chance that I wasn't yet aware of the helper methods React Native has for creating `WritableMap`s when I implemented this. I [used one of those helper methods for handling the error response (which I pass across the bridge as an object)](https://github.com/wordpress-mobile/gutenberg-mobile/blob/5ee454d143b5b588038ca11fdc4a90596685e065/react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/RNReactNativeGutenbergBridgeModule.java#L202), and I can't see why that wouldn't work for the response as well. I think it's worth revisiting whether we can switch Android to passing a successful response across the bridge as an object like we're doing with the error response. I would do that myself, but I really need to focus on the authenticated calls work first, so I created [an issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1961) to track this.

Reply:
@mkevins thanks for doing the leg work on looking up the reason for the difference. I had a todo list to come back to figure out the reasoning behind that, so I appreciate the help there :) 

@mchowning Thanks for the insight! I'll keep an eye on that issue you created. I'll see if I have time to pick that up. It would be good to tackle that soon because it doesn't look like `fetchRequest` is widely used yet.

cc @pinarol

Reply:
Thanks @mchowning for opening that issue and providing the context :+1: .

@chipsnyder glad that helped :smile: . Nice work on this block!

Maybe we can add a small comment as a reminder that this check can be removed once the `fetchRequest` is using the same types on Android and iOS, wdyt?

```suggestion
					// TODO: remove this check after `fetchRequest` types are made consist across platforms
					// (see: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1961)
					if ( typeof categoriesList === 'string' ) {
```

Reply:
Good idea I'll add that comment now.